pkgs/native_toolchain_c/lib/src/native_toolchain/msvc.dart

```diff
@@ -96,6 +102,18 @@ final Tool vcvars32 = Tool(
   ),
 );
 
+final Tool vcvarsarm64 = Tool(
+  // vcvarsarm64 only works on native windows-arm64. In case of cross
+  // compilation, it's better to stick to cross toolchain, which works under
+  // emulation on windows-arm64.
+  name: 'vcvarsamd64_arm64.bat',
```

Comment:
This has host and target arch in the bat name.

What about cross compilation, which vcvars do we use there?

Reply:
The format is `vcvars<host>_<target>.bat`, so in this case, we always use x64 -> arm64 toolchain, even on native arm64.

`vcvarsarm64.bat` script is only present on native windows-arm64.

However, cross compilation is needed when someone compiles native assets for windows-arm64, from windows-x64, which is the current work in progress I do on flutter side.
So always using x64 -> arm64 prevents having a special use case for arm64 native vs cross compiled.

Is that clearer?

Reply:
That makes sense.

I would suspect that are are downsides though by using the x64 host scripts on an arm64 host, e.g. slower compilation due to emulation.

Maybe it would be better to have all 4 versions, and select the right script based on both host and target architecture?

Reply:
https://github.com/dart-lang/native/issues/170
Added reference in a comment.